src/esrs/postgres/store.rs

```diff
@@ -0,0 +1,253 @@
+            .execute(&mut *transaction)
+            .await?;
+
+        transaction.commit().await?;
+
+        Ok(self)
+    }
+
+    /// Save an event in the event store and return a new `StoreEvent` instance.
+    pub async fn save_event(
+        &self,
+        aggregate_id: Uuid,
+        event: Manager::Event,
+        occurred_on: DateTime<Utc>,
+        sequence_number: SequenceNumber,
+        executor: impl Executor<'_, Database = Postgres>,
+    ) -> Result<StoreEvent<Manager::Event>, Manager::Error> {
+        let id: Uuid = Uuid::new_v4();
+
+        let _ = sqlx::query(self.statements.insert())
+            .bind(id)
+            .bind(aggregate_id)
+            .bind(Json(&event))
+            .bind(occurred_on)
+            .bind(sequence_number)
+            .execute(executor)
+            .await?;
+
+        Ok(StoreEvent {
+            id,
+            aggregate_id,
+            payload: event,
+            occurred_on,
+            sequence_number,
+        })
+    }
+
+    /// This function returns a stream representing the full event store table content. This should
+    /// be mainly used to rebuild read models.
+    pub fn get_all(&self) -> BoxStream<Result<StoreEvent<Manager::Event>, Manager::Error>> {
+        Box::pin({
+            sqlx::query_as::<_, event::Event>(self.statements.select_all())
+                .fetch(&self.pool)
+                .map(|res| match res {
+                    Ok(event) => event.try_into().map_err(Into::into),
+                    Err(error) => Err(error.into()),
+                })
+        })
+    }
+
+    /// This function returns the list of all projections added to this store. This function should
+    /// mostly used while creating a custom persistence flow using [`PgStore::persist`].
+    pub fn projectors(&self) -> &[Projector<Manager>] {
+        &self.projectors
+    }
+
+    /// This function returns the list of all policies added to this store. This function should
+    /// mostly used while creating a custom persistence flow using [`PgStore::persist`].
+    pub fn policies(&self) -> &[Policy<Manager>] {
+        &self.policies
+    }
+
+    /// This function could be used in order to customize the way the store persist the events.
+    /// For example could be used to avoid having projectors in transaction with event saving. Or to
+    /// let the policies return or not an error if one of them fails.
+    ///
+    /// An example of how to use this function is in `examples/customize_persistence_flow` example
+    /// folder.
+    pub async fn persist<'a, F: Send, T>(&'a self, fun: F) -> Result<Vec<StoreEvent<Manager::Event>>, Manager::Error>
+    where
+        F: FnOnce(&'a Pool<Postgres>) -> T,
+        T: Future<Output = Result<Vec<StoreEvent<Manager::Event>>, Manager::Error>> + Send,
+    {
+        fun(&self.pool).await
+    }
+
+    /// This function closes the inner pool.
+    pub async fn close(&self) {
```

Comment:
I'm not sure about having this, since the pool is a smart pointer(shared) and passed in the constructor. 
Closing it here is going to close it for the one passing the pool which is not what we want.